zarp/run/snakemake.py

```diff
@@ -0,0 +1,49 @@
+"""Module for running snakemake."""
+
+import subprocess
+import os
+
+from zarp.config.models import Run
+
+
+class SnakemakeExecutor:
+    """Run snakemake with system calls."""
```

Comment:
Please write a [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstring at least with the following sections:
- `Args`
- `Attributes`

`Args` refers to the arguments for the constructor (here: `run`), `Attributes` refers to the attributes of the class (here: `run_dict` and `success`).

Also nice to include a section `Examples`.

Here's an example for a class docstring with `Args`, `Attributes` and `Examples` (not a particularly good class, but good enough as an example for a docstring): https://github.com/zavolanlab/htsinfer/blob/bdccede0c3948e591f77f9a4112e7fef768dd1b5/htsinfer/get_library_type.py#L26-L56